### ISSUE TITLE
Constructing Key::Character(_) without importing Smolstr

### ISSUE THREAD

**Issue by dhardy** [rust-windowing/winit#2996]
Currently to construct a `Key::Character(_)` I have to:
1. Import `Smolstr`
2. Do e.g. `Key::Character(SmolStr::new_inline("/"))`

Since Winit includes `Smolstr` in its public API, I recommend that it be publically re-export (e.g. `winit::keyboard::Smolstr`) so that users do not have to add a dependency on (the correct version of) this crate too.

Further, I suggest adding convenience constructors. Some ideas:

-`fn Key::new_char(c: char) -> Option<Self>` — note that control characters are not allowed
- `fn Key::new_str(s: impl AsRef<str>) -> Option<Self>`
- `fn Key::new_str_inline(s: &str) -> Option<Self>` — uses `SmolStr::new_inline`

Since `Key::Character` is not a `char` maybe it has the wrong name? (I bet there's already far too much discussion on that topic.)

**Comment by kchibisov** [comment]
You can do `Key::Character("a".into())`, you can also use regular `&str` and compare by `str`.

**Comment by dhardy** [comment]
I don't see those impls: https://docs.rs/winit/0.29.0-beta.0/winit/keyboard/enum.Key.html#trait-implementations

**Comment by kchibisov** [comment]
The key is generic with default parameter of `SmolStr`, so into could figure out from the context. And we provide a `as_ref` to map it to `Key<&str>`.

**Comment by dhardy** [comment]
Fair enough. Re-exporting `SmolStr` may still make sense (e.g. when someone extracts the `text` from a `KeyEvent`).

**Comment by kchibisov** [comment]
Well, I don't really mind to simply reexport the crate, but it's not like you need `SmolStr` import at all.
